core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1086,6 +1109,126 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("repExactlyAs is consistent with repAs") {
+    forAll(ParserGen.gen, Gen.choose(1, Int.MaxValue), Arbitrary.arbitrary[String]) {
+      (genP, n, str) =>
+        val repA = genP.fa.repAs[NonEmptyVector[_]](n, n)
+        val repB = genP.fa.repExactlyAs[NonEmptyVector[_]](n)
+        assertEquals(repA.parse(str), repB.parse(str))
+    }
+  }
+
+  property("rep parses n entries, min <= n <= max") {
+    val validMinMax = for {
+      min <- Gen.choose(1, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
+        genP.fa.rep(min, max).parse(str).foreach { case (_, l) =>
+          assert(l.length <= max)
+          assert(l.length >= min)
+        }
+      }
+    }
+  }
+
+  property("rep0 parses n entries, min <= n <= max") {
+    val validMinMax = for {
+      min <- Gen.choose(0, Int.MaxValue)
+      max <- Gen.choose(min, Int.MaxValue)
+    } yield (min, max)
+    forAll(ParserGen.gen, validMinMax, Arbitrary.arbitrary[String]) { (genP, minmax, str) =>
+      {
+        val (min, max) = minmax
```

Comment:
This test failed to catch the max being discarded when min == 0 bug you discovered, even after running locally 10 times in a row with the bug active.

That suggests to me that this test (and the other ones that take arbitrary parsers and only may have meaningful results on success) only provide a false sense of security and a small contribution to global warming.

Shall we get rid of them? pretty much the same is tested in the rep tests on below with the "a" strings.

Reply:
I think it is fine to leave them because the law should be true. That said, I think the direct tests you added where the matches are more common are important to have as well.

Sometimes I've found bugs months after running property checks, so, I like having them.